core/frontend/src/render/FrameStats.ts

```diff
@@ -115,14 +118,19 @@ export class FrameStatsCollector {
   }
 
   public beginFrame() {
-    this._shouldRecordFrame = undefined !== this._onFrameStatsReady && this._onFrameStatsReady.numberOfListeners > 0;
+    this._shouldRecordFrame = IModelApp.renderSystem.doTelemetry || (undefined !== this._onFrameStatsReady && this._onFrameStatsReady.numberOfListeners > 0);
   }
 
   public endFrame(wasFrameDrawn = false) {
     if (this._shouldRecordFrame) {
       if (wasFrameDrawn) {
         if (undefined !== this._onFrameStatsReady)
           this._onFrameStatsReady.raiseEvent(this._frameStats); // transmit this frame's statistics to any listeners
+
+        if (IModelApp.renderSystem.doTelemetry) {
+          FrameStatsCollector._postTelemetry(this._frameStats); // eslint-disable-line @typescript-eslint/no-floating-promises

```

Comment:
@jason-crow @calebmshafer @evanpreslar-bentley
The way this is currently implemented means that IModelApp.telemetry will get data every single frame.
Display team has the perspective that any "grouping before transmitting" should be handled by the telemetry client, etc.
Any perspective on this?  Thanks.

Reply:
Sending telemetry info for every frame seems excessive to me, especially if we're sending that data over the wire from potentially every active frontend. Grouping or no grouping, that sounds like a _lot_ of data to send.

Wouldn't it make more sense to batch these calls on the renderer's side and send "heartbeat" telemetry updates regarding significant errors or events that happened during rendering?

Reply:
@evanpreslar-bentley The purpose of this feature is to gain an understanding of real performance experienced by users - not just significant errors or events.
What is the frequency of other stuff logged via this?  UIFramework, etc.

Reply:
Joe G specifically requested per-frame information. It is the display system's job to produce that information. It should **not** be the display system's job to figure out how to efficiently transmit it over the network. That sounds like the kind of thing a TelemetryManager would be responsible for.

Reply:
Understood. We don't currently batch our telemetry requests, but I'll get on that immediately.
@calebmshafer any idea what a good default batching interval would be for this? Every few seconds or so?

Reply:
@pmconne I don't know if I 100% agree. This is similar to logging, the amount of logging is always managed by the system doing the logging not the Logger itself. The telemetry client cannot predict how many times you're going to call this and the size of the payload.  It can do more intelligent batching than it currently does today, and I'll have to look but I think the AppInsights client handles this already, but that has to be in addition to managing how many requests are sent.

The UI Framework only calls this on explicit user actions (e.g. clearing overrides or emphasizing elements) or something that happens once or twice per application session (e.g. version information).

I think we may need to follow up with Joe because making, optimistically, 30-60 telemetry requests per second isn't going to work even with batching. How large would the payload be for each one? (I haven't looked the code past this comment yet)

Reply:
@evanpreslar-bentley for batching a request it looks like @NancyMcCallB has something already for that in `FeatureLogBatchClient`. That class seems to be living off on it's own right now in the frontend, I think we should deprecate/remove the existing class and make the batching capability a part of the `TelemetryManager`

Reply:
@calebmshafer 
`How large would the payload be for each one?`
See FrameStats.ts for the FrameStats interface.